core/include/usBundleResource.h

```diff
@@ -277,15 +274,21 @@ class US_Core_EXPORT BundleResource
 
 private:
 
-  BundleResource(const std::string& file, const BundleResourceContainer& resourceContainer);
-  BundleResource(int index, const BundleResourceContainer& resourceContainer);
+  BundleResource(
+        const std::string& file,
+        const std::shared_ptr<const BundleArchive>& archive
+        );
+
+  BundleResource(
+        int index,
+        const std::shared_ptr<const BundleArchive>& archive
+        );
 
-  friend class Bundle;
-  friend class BundlePrivate;
+  friend class BundleArchive;
```

Comment:
Since ````US_MSVC_PUSH_DISABLE_WARNING(4099)``` was removed this needs to be ```friend struct BundleArchive;``` otherwise a compile error occurs with VS.